File: src/test/java/com/jiro4989/tkfm/model/PositionModelTest.java
package com.jiro4989.tkfm.model;

import static org.junit.jupiter.api.Assertions.assertEquals;

import org.junit.jupiter.api.*;
import org.junit.jupiter.api.extension.ExtendWith;
import org.testfx.framework.junit5.*;

@ExtendWith(ApplicationExtension.class)
public class PositionModelTest {
  @Test
  public void testConstructor() {
    var pos = new PositionModel();
    assertEquals(0, pos.getX());
    assertEquals(0, pos.getY());
  }

  @Test
  public void testProperty() {
    var pos = new PositionModel(5, 7);
    assertEquals(5, pos.xProperty().get());
    assertEquals(7, pos.yProperty().get());

    pos.setX(10);
    pos.setY(20);
    assertEquals(10, pos.xProperty().get());
    assertEquals(20, pos.yProperty().get());
  }
}

File: src/test/kotlin/com/jiro4989/tkfm/model/PositionModelTest.kt
package com.jiro4989.tkfm.model

import org.junit.jupiter.api.*
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.extension.ExtendWith
import org.testfx.framework.junit5.*

@ExtendWith(ApplicationExtension::class)
class PositionModelTest {
  @Test
  fun testConstructor() {
    var pos = PositionModel()
    assertEquals(0.0, pos.x)
    assertEquals(0.0, pos.y)
  }

  @Test
  fun testProperty() {
    var pos = PositionModel(5.0, 7.0)
    assertEquals(5.0, pos.x)
    assertEquals(7.0, pos.y)

    pos.x = 10.0
    pos.y = 20.0
    assertEquals(10.0, pos.x)
    assertEquals(20.0, pos.y)
  }
}
